### PR TITLE
Create Azure Databricks Workbook #733

### LINKED ISSUE #733: Azure Advisor Opmization Workbook New Ideas request
## 📝 Scenario
I am part of the Finops Team here at KPMG, and we extensively use the optimization workbook for our operations. We have identified some new ideas which would provide more value to the optimization workbook and will help us fulfill our daily task much effectively.
As a **<TODO: What persona is this for?>**, I need to **<TODO: What action does the user need to perform?>** in order to **<TODO: What is the goal? Why is this needed?>**

## 💎 Solution
Ideas.

1. Flexibility to setup alerts in Optimization workbook. The goal is that we can setup email alert so that when a new orphaned resources/ recommendation is added to the workbook we can get an alert in our inbox. Right we need to manually check  everyday if any new resources are added.

2.  Cost Details.
Cost breakout by orphaned resource category: It would be a nice feature to include the cost information for the orphaned resources. Right now we need to manually check the details in the cost management tool.
Cost Anomaly 
It would be helpful if Cost Anomaly alerts were aggregated into a central view, such as a table. right need to manually check insights in each subscriptions.

3. Addition of New Services:  It would be helpful to expand the services covered in the workbook. 
Some of the services which are looking at is 
- Azure Databricks 
- Cognitive Services
- Azure Data Factory v2
- Defender for cloud (existing workbook only provide cost for different plans and total cost)
- Application insights etc.

4. Rightsizing Recommendations across services:
- SQL Database rightsizing based on current utilization. Right now we can see we have only Idle SQL databases
- Insights about SQL Database retention policies, similar to Recovery Services Vault in Backup Center
- Managed disk capacity under-utilization to reduce cost
- Rightsizing recommendations for minimum Autoscale instances for App service plans, Application gateway based on current utilization
- Premium File shares underutilized provisioned capacity
-  Storage Account Access tier change for Blobs based on access pattern
- App service Plan rightsizing
- Azure Synapse Analytics rightsizing
- Azure SQL Managed instances rightsizing
- Azure Redis Cache rightsizing

5. Idle instances addition
- Idle VPN gateways 
- Unhealthy Backend pools for App gateway
- Idle VM for last 3months
- Storage not accessed for last 3months
- SQL Elastic Pools without DB's
- Disk Attached to stopped instances
- idle Load Balances with Low traffic

Please ignore if you see any ideas which are already implemented and duplicated.

## ℹ️ Additional context
TODO: Add any other context or screenshots about the change request, including alternative solutions you considered and why you ruled them out.

## 🙋‍♀️ Ask for the community

We could use your help:
1. Please vote this issue up (👍) to prioritize it.
8. Leave comments to help us solidify the vision.

### PR DESCRIPTION
**Pull request by ChiaChun0**
## 🛠️ Description
This update introduces a new Azure Databricks workbook focused on improving data insights. The workbook highlights configuration resources and offers best practice recommendations to optimize performance and align with the document (https://learn.microsoft.com/en-us/azure/databricks/lakehouse-architecture/cost-optimization/best-practices).

Fixes #733 - Azure Advisor Optimization Workbook New Ideas request

## 📷 Screenshots
- Tags for Databricks workspaces
Tags for Databricks workspaces

- Use the right instance type for Databricks clusters
Use the right instance type for Databricks clusters

- Use up-to-date runtimes for workloads
Use up-to-date runtimes for workloads

- Use auto-scaling compute for clusters
Use auto-scaling compute for clusters

- Use auto termination for clusters
Use auto termination for clusters

- Use SQL warehouse or serverless services for SQL workloads
Use SQL warehouse or serverless services for SQL workloads

### 🔬 How did you test this change?

> - [ ] 🤏 Lint tests
> - [ ] 🤞 PS -WhatIf / az validate
> - [x] 👍 Manually deployed + verified
> - [ ] 💪 Unit tests
> - [ ] 🙌 Integration tests

### 🙋‍♀️ Do any of the following that apply?

> - [ ] 🚨 This is a breaking change.
> - [ ] 🤏 The change is less than 20 lines of code.

### 📑 Did you update `docs/changelog.md`?

> - [ ] ✅ Updated changelog (required for `dev` PRs)
> - [x] ➡️ Will add log in a future PR (feature branch PRs only)
> - [ ] ❎ Log not needed (small/internal change)

### 📖 Did you update documentation?

> - [ ] ✅ Public docs in `docs` (required for `dev`)
> - [ ] ✅ Internal dev docs in `src` (required for `dev`)
> - [x] ➡️ Will add docs in a future PR (feature branch PRs only)
> - [ ] ❎ Docs not needed (small/internal change)

### CODE REVIEW

**Comment by ChiaChun0** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: src/workbooks/optimization/Azure Databricks.workbook
Comment (flanakin) [inline src/workbooks/optimization/Azure Databricks.workbook:1]: Is this intended to be part of the optimization workbook or its own independent workbook dedicated to Databricks?

**Review by flanakin — DISMISSED** [review]

File: src/workbooks/optimization/Azure Databricks.workbook
[the remark this answers is not in the document]
Comment (ChiaChun0) [reply src/workbooks/optimization/Azure Databricks.workbook:1]: Hi Flanakin, 
Sorry for missing this message.

This is intended to be part of the optimization workbook - will update it.
Thank you.

**Comment by arthurclares** [comment]
I'm merging this PR to the Workbook branch to keep developing this further. This workbook will eventually be merged to COW.

**Review by arthurclares — APPROVED** [review]
Merging this to a feature branch to continue development. This won't go into production right now.

**Pull request merged**